src/audio_core/cubeb_input.cpp

```diff
@@ -144,6 +144,11 @@ Samples CubebInput::Read() {
     while (impl->sample_queue.Pop(queue)) {
         samples.insert(samples.end(), queue.begin(), queue.end());
     }
+
+    if (samples.empty()) {
```

Comment:
What if the buffer is not empty, but different to the amount of samples the game "expects"?

Reply:
Good question, I haven't tested any other game that could theoretically cause that ? Either way the function wouldn't be called if it isn't empty.

Reply:
Can you elaborate more on the second sentence?

Reply:
The goal of this commit is specifically to return some "silent" data to the app if it is currently sampling, just like a real 3ds, there is no hardware mic cut off. I rely on all the other checks and errors that are catched (see the *_input.cpp files) does if there is an error or mistake in the sinks, it will be catched, and if everything is right but somehow the samples are empty, it will instead return silent data.

Now regarding the possible buffer being different to the amount of samples the game expects, I haven't seen such occurrence. Besides, handling the supposedly different audio data buffer is out of scope of this PR as samples actually do contain data, whether that buffer size is what the app expects or not should be handled in a different PR.